### PR TITLE
Fix #2268: SNTP Assets Download Improvements.

### LINKED ISSUE #2268: SNTP not showing when disabled, then enabled on app launch
### Description: 
No SNTP image is shown when toggle the `Show sponsored images` switch 

### Steps to Reproduce 
  1. Clean profile 1.14.3
  2. Enable rewards through on-boarding
  3. Open SNTP saw sponsored image once 
  4. Navigate to settings->New Tab page->Turn off `Show sponsored images`
  5. Remove the app from memory
  6. Relaunch the app and navigate to settings->New Tab page->Turn on `show sponsored images`
Actual: No sponsored images are shown.

**Actual result:** 
No SNTP image is shown when toggle the `Show sponsored images` switch to ON

**Expected result:**
SNTP images should be shown when `Show sponsored images`switch is ON

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Always

**Brave Version:** 

1.14.3 (20.01.28.18)
**Device details:** 
iPhone 8 - IOS 13.3

**Website problems only:**
- did you check with Brave Shields down? NA
- did you check in Safari/Firefox (WkWebView-based browsers)? NA

### Additional Information
cc: @brave/legacy_qa @jhreis @davidtemkin @anthonypkeane @jamesmudgett

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes issue #2268 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by iccub** [comment]
Do not merge, wait for @jhreis decision if we want to check for NTP stuff on every launch

**Review by jhreis — CHANGES_REQUESTED** [review]
I think this is too aggressive, this would attempt to re-fetch assets on every launch.

Assets aren't fetched when sponsored images are disabled (and changing the timing won't impact this, the network request isn't even attempted), so instead we should wipe the timer when the user re-enables (or disables) SNTP.

**Comment by iccub** [comment]
@jumde are you okay with adding a reset on every app launch so it will attempt to download a new resource if it exists? This is what we currently do for adblock resources

So is this ok from security perspective:
- on app launch attempt to download assets
- if app isn't killed(foreground or bg) and after 5hours + the random component we try to download resources again

File: Client/Frontend/Browser/HomePanel/NewTabPageBackgroundDataSource.swift [hunk 1]
```diff
@@ -124,7 +116,6 @@ class NewTabPageBackgroundDataSource {
             let attemptSponsored = Preferences.NewTabPage.backgroundSponsoredImages.value
                 && backgroundRotationCounter == NewTabPageBackgroundDataSource.sponsorshipShowValue
                 && !PrivateBrowsingManager.shared.isPrivateBrowsing
-                && BraveAds.isCurrentLocaleSupported()
```
Comment (jhreis) [inline Client/Frontend/Browser/HomePanel/NewTabPageBackgroundDataSource.swift:127]: Client's shouldn't be checking for this, this is up to the server.

Comment (iccub) [reply Client/Frontend/Browser/HomePanel/NewTabPageBackgroundDataSource.swift:127]: Sounds good to me

**Comment by iccub** [comment]
Approving Joel's changes

@jhreis please approve by yourself and merge after CI completes

**Review by jhreis — APPROVED** [review]

**Pull request merged**